pkg/kubectl/cmd/cp_test.go

```diff
@@ -554,6 +556,7 @@ func TestCopyToPod(t *testing.T) {
 	}
 
 	for name, test := range tests {
+		opts := NewCopyOptions(genericclioptions.IOStreams{Out: buf, ErrOut: errBuf})
```

Comment:
There's `genericclioptions.NewTestIOStreams` which returns the iostreams and the buffers behind it, perfect for what you need here.